vms/platformvm/block/builder/builder.go

```diff
@@ -103,24 +102,14 @@ func New(
 
 // AddUnverifiedTx verifies a transaction and attempts to add it to the mempool
 func (b *builder) AddUnverifiedTx(tx *txs.Tx) error {
-	if !b.txExecutorBackend.Bootstrapped.Get() {
-		return ErrChainNotSynced
-	}
-
 	txID := tx.ID()
 	if b.Mempool.Has(txID) {
 		// If the transaction is already in the mempool - then it looks the same
 		// as if it was successfully added
 		return nil
 	}
 
-	verifier := txexecutor.MempoolTxVerifier{
-		Backend:       b.txExecutorBackend,
-		ParentID:      b.blkManager.Preferred(), // We want to build off of the preferred block
-		StateVersions: b.blkManager,
-		Tx:            tx,
-	}
-	if err := tx.Unsigned.Visit(&verifier); err != nil {
+	if err := b.blkManager.VerifyTx(tx); err != nil {
```

Comment:
Rename `VerifyTx` to something like `VerifyMempoolCandidate` both here and in X-chain.
This way we can avoid confusion and minimize awkwardness of having to state verify mempool candidate in blkManager. We can fully solve the awkwardness later on and jontly across P-chain and X-chain